src/layouts/Resources.jsx

```diff
@@ -1,378 +1,166 @@
-  }
-
-  onFileClick = (path) => {
-    this.setState({
-      filePath: path,
-      isSelectingFile: false,
-    });
-  }
-
-  stageFileForUpload = (fileName, fileData) => {
-    const { type } = this.props;
-    const baseFolder = type === 'file' ? 'files' : 'images';
-    this.setState({
-      isFileStagedForUpload: true,
-      stagedFileDetails: {
-        path: `${baseFolder}%2F${fileName}`,
-        content: fileData,
-        fileName,
-      },
-    });
-  }
-
-  readFileToStageUpload = async (event) => {
-    const fileReader = new FileReader();
-    const fileName = event.target.files[0].name;
-    fileReader.onload = (() => {
-      /** Github only requires the content of the file
-         * fileReader returns  `data:application/pdf;base64, {fileContent}`
-         * hence the split
-         */
-
-      const fileData = fileReader.result.split(',')[1];
-      this.stageFileForUpload(fileName, fileData);
-    });
-    fileReader.readAsDataURL(event.target.files[0]);
-    this.toggleFileModal()
-  }
-
-  settingsToggle = (event) => {
-    const { id } = event.target;
-    const idArray = id.split('-');
-    let resourcePageIndex = idArray[1];
-
-    // User clicked on the "Create New Resource" button
-    if (resourcePageIndex === 'NEW') {
-      this.setState({
-        settingsIsActive: true,
-        selectedResourcePage: {
-          isNewPost: true,
-        },
-      });
-    } else if (resourcePageIndex === 'CLOSE') {
-      // User clicked on the close button in the settings modal
-      this.setState({ settingsIsActive: false });
-    } else if (resourcePageIndex === '') {
-      // User clicked on the file upload button
-      this.setState({
-        settingsIsActive: false,
-        isSelectingFile: true
-       });
-    } else {
-      // User clicked on the settings icon on an existing resource
-      resourcePageIndex = parseInt(resourcePageIndex, RADIX_PARSE_INT);
-
-      this.setState((currState) => ({
-        settingsIsActive: true,
-        selectedResourcePage: {
-          isNewPost: false,
-          category: currState.resourcePages[resourcePageIndex].category,
-          fileName: currState.resourcePages[resourcePageIndex].fileName,
-        },
-      }));
-    }
-  }
+  }, [])
 
-  changeHandler = (event) => {
+  const changeHandler = (event) => {
     const { value } = event.target;
-    this.setState({ newResourceRoomName: value });
+    setNewResourceRoomName(value)
   }
 
-  createResourceRoom = async () => {
+  const createResourceRoom = async () => {
     try {
-      const { match } = this.props;
       const { siteName } = match.params;
-      const { newResourceRoomName } = this.state;
       const params = { resourceRoom: newResourceRoomName };
-      await axios.post(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/resource-room`, params, {
-        withCredentials: true,
-      });
+      await axios.post(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/resource-room`, params);
```

Comment:
Either reload the page after the resource room is created, or use a hook so that the page is re-rendered. Currently, after a room is created, nothing happens on screen and there is no user feedback

Reply:
Resolved in https://github.com/isomerpages/isomercms-frontend/pull/207/commits/0571ba015d848b32415ba10447d4753fdbeabccb